Without this change we get this error:

```
18:09:36 clients installdeps: -r/var/lib/jenkins/tools/0/charm-test-infra/py3-clients-requirements.txt
18:11:30 clients installed: You are using pip version 8.1.1, however version 20.3.1 is available.,You should consider upgrading via the 'pip install --upgrade pip' command.,aodhclient==2.1.1,appdirs==1.4.4,argcomplete==1.12.2,async-generator==1.10,attrs==20.3.0,Babel==2.9.0,bcrypt==3.2.0,boto3==1.16.29,botocore==1.19.29,certifi==2020.11.8,cffi==1.14.4,chardet==3.0.4,cliff==3.5.0,cmd2==1.4.0,colorama==0.4.4,cryptography==3.2.1,debtcollector==2.2.0,decorator==4.4.2,dnspython==2.0.0,fasteners==0.15,futurist==1.10.0,gnocchiclient==7.0.7,hvac==0.6.4,idna==2.10,importlib-metadata==3.1.1,importlib-resources==3.3.0,ipaddress==1.0.23,iso8601==0.1.13,Jinja2==2.11.2,jmespath==0.10.0,jsonpatch==1.28,jsonpointer==2.0,jsonschema==3.2.0,juju==2.8.4,juju-deployer==0.11.0,juju-wait==2.8.4,jujubundlelib==0.5.6,jujuclient==0.54.0,jujucrashdump==0.0.0,keystoneauth1==4.3.0,lxml==4.6.2,macaroonbakery==1.3.1,MarkupSafe==1.1.1,mojo==0.4.5,monotonic==1.5,msgpack==1.0.0,munch==2.5.0,mypy-extensions==0.4.3,ndg-httpsclient==0.3.3,netaddr==0.8.0,netifaces==0.10.9,openstacksdk==0.52.0,os-client-config==2.1.0,os-service-types==1.7.0,osc-lib==2.3.0,oslo.concurrency==4.3.1,oslo.config==6.11.3,oslo.context==3.1.1,oslo.i18n==5.0.1,oslo.log==4.4.0,oslo.serialization==4.0.1,oslo.utils==4.7.0,osprofiler==3.4.0,packaging==20.7,paramiko==2.7.2,pbr==5.5.1,pika==1.1.0,pkg-resources==0.0.0,prettytable==0.7.2,protobuf==3.14.0,pyasn1==0.4.8,pycparser==2.20,pyinotify==0.9.6,pylxd==2.0.7,pymacaroons==0.13.0,PyNaCl==1.4.0,pyOpenSSL==20.0.0,pyparsing==2.4.7,pyperclip==1.8.1,pyRFC3339==1.1,pyrsistent==0.17.3,python-barbicanclient==5.0.1,python-ceilometerclient==2.9.0,python-cinderclient==7.2.0,python-codetree==1.2.0,python-dateutil==2.8.1,python-designateclient==2.12.0,python-glanceclient==3.2.2,python-heatclient==2.3.0,python-ironicclient==4.4.0,python-keystoneclient==4.2.0,python-manilaclient==1.29.0,python-neutronclient==7.2.1,python-novaclient==17.2.1,python-octaviaclient==1.10.1,python-openstackclient==5.4.0,python-swiftclient==3.11.0,pytz==2020.4,PyYAML==5.3.1,requests==2.25.0,requests-unixsocket==0.2.0,requestsexceptions==1.4.0,rfc3986==1.4.0,s3transfer==0.3.3,simplejson==3.17.2,six==1.15.0,stevedore==3.3.0,tenacity==6.2.0,theblues==0.5.2,toposort==1.5,typing-extensions==3.7.4.3,typing-inspect==0.6.0,ujson==4.0.1,UNKNOWN==0.0.0,urllib3==1.26.2,warlock==1.3.3,wcwidth==0.2.5,WebOb==1.8.6,websocket-client==0.57.0,websocket-client-py3==0.15.0,websockets==7.0,wrapt==1.12.1,ws4py==0.5.1,zaza==0.0.2.dev1,zaza.openstack==0.0.1.dev1,zipp==3.4.0
18:11:30 clients run-test-pre: PYTHONHASHSEED='0'
18:11:30 clients run-test: commands[0] | mojo --version
18:11:32 /var/lib/jenkins/tools/0/charm-test-infra/.tox/clients/lib/python3.5/site-packages/OpenSSL/crypto.py:14: CryptographyDeprecationWarning: Python 3.5 support will be dropped in the next release of cryptography. Please upgrade your Python.
18:11:32   from cryptography import utils, x509
18:11:32 0.4.5
18:11:32 clients run-test: commands[1] | openstack --version
18:11:32 Traceback (most recent call last):
18:11:32   File "/var/lib/jenkins/tools/0/charm-test-infra/.tox/clients/lib/python3.5/site-packages/netaddr/compat.py", line 91, in <module>
18:11:32     from importlib import resources as _importlib_resources
18:11:32 ImportError: cannot import name 'resources'
```

http://osci:8080/job/test_charm_single/119569/console

--- a/py3-clients-requirements.txt
+++ b/py3-clients-requirements.txt
@@ -25,27 +25,37 @@ babel!=2.4.0,>=2.3.4
 paramiko  # For mojo_os_utils
 
 # OpenStack
-# Newer versions dropped Python 3.5 support:
-python-openstackclient>=3.14.0,<3.19.0
-aodhclient<1.3.0
-python-designateclient>=1.5,<3.0.0
-python-barbicanclient<5.0.0
+python-openstackclient>=3.14.0,<3.19.0; python_version < '3.6'
+python-openstackclient>=3.14.0; python_version >= '3.6'
+aodhclient<1.3.0; python_version < '3.6'
+aodhclient; python_version >= '3.6'
+python-designateclient>=1.5,<3.0.0; python_version < '3.6'
+python-designateclient>=1.5; python_version >= '3.6'
+python-barbicanclient<5.0.0; python_version < '3.6'
+python-barbicanclient; python_version >= '3.6'
 python-ceilometerclient
-python-cinderclient<4.3.0
-python-glanceclient<2.17.0
-python-heatclient<1.18.0
-python-keystoneclient<3.20.0
-python-neutronclient<6.13.0
-python-novaclient<14.0.0
-python-swiftclient<=3.8.0
+python-cinderclient<4.3.0; python_version < '3.6'
+python-cinderclient; python_version >= '3.6'
+python-glanceclient<2.17.0; python_version < '3.6'
+python-glanceclient; python_version >= '3.6'
+python-heatclient<1.18.0; python_version < '3.6'
+python-heatclient; python_version >= '3.6'
+python-keystoneclient<3.20.0; python_version < '3.6'
+python-keystoneclient; python_version >= '3.6'
+python-neutronclient<6.13.0; python_version < '3.6'
+python-neutronclient; python_version >= '3.6'
+python-novaclient<14.0.0; python_version < '3.6'
+python-novaclient; python_version >= '3.6'
+python-swiftclient<=3.8.0; python_version < '3.6'
+python-swiftclient; python_version >= '3.6'
 
 # Newer versions use keywords that didn't exist in
 # python 3.5 yet (e.g. "ModuleNotFoundError")
-importlib-metadata<3.0.0
-importlib-resources<3.0.0
+importlib-metadata<3.0.0; python_version < '3.6'
+importlib-resources<3.0.0; python_version < '3.6'
 
 # Missing dependency of python-openstackclient:
-dogpile.cache<1.0.0
+dogpile.cache<1.0.0; python_version < '3.6'
 
 # Dependency of python-glanceclient. Newer versions dropped Python 3.5 support:
-oslo.utils<=3.41.0
+oslo.utils<=3.41.0; python_version < '3.6'

